notebooks/M4_notebook.ipynb

```diff
@@ -0,0 +1,219 @@
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```python\n",
+    "from flask import Flask, request, jsonify\n",
+    "import joblib\n",
+    "\n",
+    "import numpy as np\n",
+    "\n",
+    "app = Flask(__name__)\n",
+    "\n",
+    "# 1. Load your model here\n",
+    "model = joblib.load(\"model.joblib\")\n",
+    "\n",
+    "# 2. Define a prediction function\n",
+    "def return_prediction(data):\n",
+    "\n",
+    "    # format input_data here so that you can pass it to model.predict()\n",
+    "\n",
+    "    return model.predict(np.array([data]))\n",
+    "\n",
+    "# 3. Set up home page using basic html\n",
+    "@app.route(\"/\")\n",
+    "def index():\n",
+    "    # feel free to customize this if you like\n",
+    "    return \"\"\"\n",
+    "    <h1>Welcome to our rain prediction service</h1>\n",
+    "    To use this service, make a JSON post request to the /predict url with 5 climate model outputs.\n",
+    "    \"\"\"\n",
+    "\n",
+    "# 4. define a new route which will accept POST requests and return model predictions\n",
+    "@app.route('/predict', methods=['POST'])\n",
+    "def rainfall_prediction():\n",
+    "    content = request.json  # this extracts the JSON content we sent\n",
+    "    inputs = content[\"data\"]\n",
+    "    prediction = return_prediction(inputs)\n",
+    "    results = {\"Input\": inputs, \"Prediction\": prediction.tolist()}  # return whatever data you wish, it can be just the prediction\n",
+    "                     # or it can be the prediction plus the input data, it's up to you\n",
+    "    return jsonify(results)\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<hr>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 2. Deploy your API"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Screenshot**"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "![](../img/m4_task_2.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "> Please note that our group created a new EC2 instance (Owner: student84) for task 2 deployment of our API."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<hr>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 3. Summarize your journey from Milestone 1 to Milestone 4"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Throughout these 4 milestones, our group sequentially took on the roles of 1. Data Engineer, 2. Infrastructure, 3. Data Scientist, and 4. DevOps to perform prediction tasks on the large `daily rainfall in Australia` [dataset](https://figshare.com/articles/dataset/Daily_rainfall_over_NSW_Australia/14096681) (5.7 GB), and we have achieved the following 4 objectives in each of the milestones.**"
```

Comment:
```suggestion
    "**Throughout these 4 milestones, our group sequentially took on the roles of 1. Data Engineer, 2. Infrastructure Engineer, 3. Data Scientist, and 4. DevOps to perform prediction tasks on the large `daily rainfall in Australia` [dataset](https://figshare.com/articles/dataset/Daily_rainfall_over_NSW_Australia/14096681) (5.7 GB), and we have achieved the following 4 objectives in each of the milestones.**"
```

Reply:
Some minor suggestion to change the name of our role :)

Reply:
Thank you for the catch and great suggestion!